#! /bin/sh /usr/share/dpatch/dpatch-run
## 82_prefs.dpatch by Mike Hommey <glandium@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Set javascript.options.showInConsole.
## DP: Set DPI to system settings.
## DP: Fix font defaults.

@DPATCH@

=== modules/libpref/src/init/all.js
==================================================================
--- xulrunner/modules/libpref/src/init/all.js	(revision 67)
+++ xulrunner/modules/libpref/src/init/all.js	(local)
@@ -439,6 +439,7 @@
 pref("javascript.enabled",                  true);
 pref("javascript.allow.mailnews",           false);
 pref("javascript.options.strict",           false);
+pref("javascript.options.showInConsole",    true);
 
 // advanced prefs
 pref("security.enable_java",                true);
@@ -1844,7 +1844,7 @@
 pref("network.hosts.smtp_server", "localhost");
 pref("network.hosts.pop_server", "pop");
 pref("network.protocol-handler.warn-external.file", false);
-pref("layout.css.dpi", -1); // max(96dpi, System setting)
+pref("layout.css.dpi", 0); // System setting
 pref("browser.drag_out_of_frame_style", 1);
 pref("editor.singleLine.pasteNewlines", 0);

@@ -1872,33 +1873,33 @@
 
 // th
 
-pref("font.name.serif.tr", "Times");
-pref("font.name.sans-serif.tr", "Helvetica");
-pref("font.name.monospace.tr", "Courier");
+pref("font.name.serif.tr", "serif");
+pref("font.name.sans-serif.tr", "sans-serif");
+pref("font.name.monospace.tr", "monospace");
 
 pref("font.name.serif.x-baltic", "serif");
 pref("font.name.sans-serif.x-baltic", "sans-serif");
 pref("font.name.monospace.x-baltic", "monospace");
 
-pref("font.name.serif.x-central-euro", "Times");
-pref("font.name.sans-serif.x-central-euro", "Helvetica");
-pref("font.name.monospace.x-central-euro", "Courier");
+pref("font.name.serif.x-central-euro", "serif");
+pref("font.name.sans-serif.x-central-euro", "sans-serif");
+pref("font.name.monospace.x-central-euro", "monospace");
 
 pref("font.name.serif.x-cyrillic", "serif");
 pref("font.name.sans-serif.x-cyrillic", "sans-serif");
 pref("font.name.monospace.x-cyrillic", "monospace");
 
-pref("font.name.serif.x-unicode", "Times");
-pref("font.name.sans-serif.x-unicode", "Helvetica");
-pref("font.name.monospace.x-unicode", "Courier");
+pref("font.name.serif.x-unicode", "serif");
+pref("font.name.sans-serif.x-unicode", "sans-serif");
+pref("font.name.monospace.x-unicode", "monospace");
 
-pref("font.name.serif.x-user-def", "Times");
-pref("font.name.sans-serif.x-user-def", "Helvetica");
-pref("font.name.monospace.x-user-def", "Courier");
+pref("font.name.serif.x-user-def", "serif");
+pref("font.name.sans-serif.x-user-def", "sans-serif");
+pref("font.name.monospace.x-user-def", "monospace");
 
-pref("font.name.serif.x-western", "Times");
-pref("font.name.sans-serif.x-western", "Helvetica");
-pref("font.name.monospace.x-western", "Courier");
+pref("font.name.serif.x-western", "serif");
+pref("font.name.sans-serif.x-western", "sans-serif");
+pref("font.name.monospace.x-western", "monospace");
 
 pref("font.name.serif.zh-CN", "serif");
 pref("font.name.sans-serif.zh-CN", "sans-serif");
